### DISCUSSION TITLE
SA update

### DISCUSSION THREAD

**Discussion by hwz9612** [strongswan/strongswan#1038] [Q&A]
Hi,
   Why does the SPI update after a while？ How do we configure strongswan to have the SPI update at a specified time？
Thanks.

**Comment by tobiasbrunner** [comment]
Please read [this](https://docs.strongswan.org/docs/5.9/config/rekeying.html).

**Reply by hwz9612** [reply]
Hi,
   Now I configure strongswan in ipsec.conf.
conn windows7
    lifetime=24h
    keyexchange=ikev2
    ike=aes256-sha1-modp1024!
    esp=aes256-sha1!
    left=%defaultroute
    leftid=192.18.18.225
    leftauth=pubkey
    leftsubnet=0.0.0.0/0
    leftcert=server.cert.pem
    right=%any
    rightauth=eap-mschapv2
    rightsourceip=10.31.2.0/24
    rightsendcert=never
    eap_identity=%any
    auto=add
After the windows7 client connected to strongswan, I transfered file between the windows7 client and the target server. When a file of a certain size is transferred, the SPI between the windows7 client and strongswan is updated. The rekey time did not reach the set 24 hours.
Besides I set lifebytes and lifepackets, but it is not useful. When a file of a certain size is transferred, the SPI between the windows7 client and strongswan is still updated.
Why does this happen？

**Reply by tobiasbrunner** [reply]
That's because Windows rekeys IPsec SAs based on the number transferred bytes (I don't remember the exact limit but I think it's every few hundred MBs). These limits are not negotiated with IKEv2, so it's up to each peer when they want to replace the SAs.

**Reply by hwz9612** [reply]
Hi,
Thanks for your explanation. How do I configure windows so that the windows client does not rekey IPsec SAs?
Thanks.

**Reply by tobiasbrunner** [reply]
I'm not aware of any such option.
